src/stubs/ES6Map.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright 2015, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ES6Map
+ */
+
+'use strict';
+
+function cloneWithoutKey(obj, notKey) {
+  var newObj = {};
+  var keys = Object.keys(obj);
+  for (var i = 0, l = keys.length; i < l; i++) {
+    var key = keys[i];
+    if (key !== notKey) {
+      newObj[key] = obj[key];
+    }
+  }
+  return newObj;
+}
+
+if (typeof Map !== 'undefined') {
+  var _Map = Map;
+   _Map.remove = _Map.delete;
+  module.exports = _Map;
+}
+else {
+  var ES6Map = function ES6Map() {
+    this.values = {};
+  };
+  ES6Map.prototype = {
+
+    remove: function(key) {
+      this.values = cloneWithoutKey(this.values, key);
```

Comment:
Just trying to implement a todo :-). I would expect es6 maps (and sets also) to be useful in more places, so that's why the extra file is involved. For the crappy polyfill, well that's because I was experimenting a bit - and thought it would be better to throw in the open. Very much aware it was crap though (sorry though :P). So yeah, therefore my questions - could we use the ReactInstanceMap 'map polyfill'? Or, also possible: no it's better to duplicate code in this case.